sharedPrompts/REFACTORING_GUIDE.md

```
@@ -0,0 +1,297 @@
+**구현 위치**: `PaymentEventPublisherAdapter`에서 이미 `ApplicationEventPublisher` 사용 중
+
+### 2.2 Payment 엔티티 메서드 호출
+
+**문제**: `payment.approve()` 메서드 호출 형식 불일치
+
+**기존 구현**:
+```java
+public void approve(String externalPaymentId) {
+    markSuccess(externalPaymentId);
+}
+
```

markdownlint-cli2 (0.21.0)

[warning] Tables should be surrounded by blank lines

(MD058, blanks-around-tables)